frame/bribedao/src/lib.rs

```diff
@@ -0,0 +1,315 @@
+		fn delete_bribe(request: DeleteBribeRequest) -> Result<bool, DispatchError> {
+			// todo, make sure this function can not be abused
+			Self::do_delete_bribe(request)
+			//			Ok(true)
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Create a new Bribe Request
+		fn do_create_bribe(request: CreateBribeRequest<T>) -> Result<BribeIndex, DispatchError> {
+			let id = BribeCount::<T>::mutate(|id| {
+				*id += 1;
+				*id
+			});
+
+			ensure!(!BribeRequests::<T>::contains_key(id), Error::<T>::AlreadyBribed); //dont duplicate briberequest if we already have it
+
+			BribeRequests::<T>::insert(id, request);
+			Ok(id)
+		}
+
+		/// Find votes for a bribe request
+		fn do_match_votes(bribe_index: BribeIndex) -> Result<bool, DispatchError> {
+			ensure!(BribeRequests::<T>::contains_key(bribe_index), Error::<T>::InvalidIndex);
+
+			let bribe_request = BribeRequests::<T>::get(bribe_index).unwrap();
+
+			let ref_index = bribe_request.ref_index;
+			// Yield all the bribe votes for sale with the same ref index
+
+			let loot: Vec<BribesStorage> = Fastvec::<T>::get().find_all_pid(ref_index);
+
+			if !loot.is_empty() {
+				let mut spendamount: u32 = 0;
+
+				for bribes in loot {
+					// Cast Vote
+
+					let vote = Vote { aye: bribe_request.is_aye, conviction: Default::default() }; //todo
+
+					let ss = bribe_request.clone();
+					T::Democracy::vote(ss.account_id, bribes.p_id, vote); //AccountId,
+
+					// Remove from storage
+					Fastvec::<T>::mutate(|a| {
+						a.remove_bribe(bribes.amount, bribes.p_id, bribes.votes)
+					});
+
+					// append the amount to our spend amount tracking
+					spendamount += bribes.amount;
+
+					// Pay out to the seller of the vote
+
+					let tmp_value = bribe_request.clone();
+					let currencyid = tmp_value.asset_id;
+					T::Currency::release(
+						currencyid,
+						&tmp_value.account_id,
+						bribes.amount.into(),
+						false,
+					)
+					.map_err(|_| Error::<T>::ReleaseFailed)?;
+					// todo: Check if all votes are fullfilled
+					let bribe_balance: u32 = bribe_request.total_reward.saturated_into::<u32>();
+					//					if we have spent all the money we have for votes, we assume the order is
+					// fullfilled and can not interact anymore so we remove it
+					if spendamount >= bribe_balance {
+						//todo also check if the correct amount of votes has been fullfilled
+						// Delete The bribe if fullfilled
+						let dr: DeleteBribeRequest = DeleteBribeRequest { bribe_index };
+						Self::do_delete_bribe(dr).map_err(|_| Error::<T>::BribeDeletionFailed)?;
+					}
+				}
+			}
+			Ok(true)
+		}
+
+		/// Take Bribe user sell votes request   
+		fn do_take_bribe(request: TakeBribeRequest<T>) -> Result<bool, DispatchError> {
+			ensure!(
+				BribeRequests::<T>::contains_key(request.bribe_index),
+				Error::<T>::InvalidIndex
+			);
+			// todo: make sure the user is not selling the same vote twice
+
+			let bribe_request = BribeRequests::<T>::get(request.bribe_index).unwrap();
```

Comment:
Similar to previous comment, we can merge this two statements. 
Also, it look like the function `do_match_votes` is also looking up for the request can't we avoid this double lookup by directly passing the request to `do_match_votes` in this function?

Reply:
fixed in latest